packages/perseus/src/widgets/numeric-input/numeric-input-validator.ts

```diff
@@ -107,62 +108,48 @@ function numericInputValidator(
     // If `currentValue` is not TeX, this should be a no-op.
     const currentValue = ParseTex(userInput.currentValue);
-    const correctAnswers = rubric.answers.filter(
-        (answer) => answer.status === "correct",
-    );
-
-    const normalizedAnswerExpected = correctAnswers.every(
-        (answer) => Math.abs(answer.value) <= 1,
-    );
-
-    // Look through all correct answers for one that matches either
-    // precisely or approximately and return the appropriate message:
-    // - if precise, return the message that the answer came with
-    // - if it needs to be simplified, etc., show that message
-    let result = correctAnswers
+
+    const normalizedAnswerExpected = rubric.answers
+        .filter((answer) => answer.status === "correct")
+        .every((answer) => Math.abs(answer.value) <= 1);
+
+    // The coefficient is an attribute of the widget
+    let localValue: string | number = currentValue;
+    if (rubric.coefficient) {
+        if (!localValue) {
+            localValue = 1;
+        } else if (localValue === "-") {
+            localValue = -1;
+        }
+    }
+    const matchedAnswer:
+        | (PerseusNumericInputAnswer & {score: Score})
+        | undefined = rubric.answers
         .map((answer) => {
-            // The coefficient is an attribute of the widget
-            let localValue: string | number = currentValue;
-            if (rubric.coefficient) {
-                if (!localValue) {
-                    localValue = 1;
-                } else if (localValue === "-") {
-                    localValue = -1;
-                }
-            }
-            const validate = createValidator(answer);
-            return validate(
+            const validateFn = createValidator(answer);
+            const score = validateFn(
                 maybeParsePercentInput(localValue, normalizedAnswerExpected),
             );
+            return {...answer, score};
         })
-        .find((match) => match.correct || match.empty);
-
-    if (!result) {
-        // Otherwise, if the guess is not correct
-        const otherAnswers = [].concat(
-            // @ts-expect-error - TS2769 - No overload matches this call.
-            rubric.answers.filter((answer) => answer.status === "ungraded"),
-            rubric.answers.filter((answer) => answer.status === "wrong"),
-        );
-
-        // Look through all other answers and if one matches either
-        // precisely or approximately return the answer's message
-        const match = otherAnswers.find((answer) => {
-            const validate = createValidator(answer);
-            return validate(
-                maybeParsePercentInput(currentValue, normalizedAnswerExpected),
-            ).correct;
+        .find((answer) => {
+            // NOTE: "answer.score.correct" indicates a match via the validate function.
+            //       It does NOT indicate that the answer itself is correct.
+            return (
```

Comment:
This is a very helpful comment, thank you! I suspect the validators were made before Expression/Numeric were updated to allow for the inclusion of "wrong" or "ungraded" answers.